ml-image/scripts/evaluation_annotation.py

```diff
@@ -93,24 +98,143 @@ def label_results(detections: list, subframes: list, subframe_indices: list) ->
+        f'Outputting frame-numbered trailer to {trailer_path}_with_frames.avi')
+    if os.path.isfile(f'{trailer_path}_with_frames.avi'):
+        print(f'{trailer_path}_with_frames.avi already exists, moving on')
     else:
-        get_frames(trailer_path)
+        get_frames(f'{trailer_path}.mp4')
     indices = list(
         map(int, input("Enter frame indices to annotate separated by space: ").split()))
     print(f'Annotating frames {indices}...')
-    visiondetector = get_vision_module()
-    subframes = sample_frames(indices, trailer_path)
-    detections = get_faces(subframes, visiondetector)
-    d_labeled_results = label_results(detections, subframes, indices)
-    with open(f'annotated_{trailer_path[:-4]}.pkl', 'wb') as outfile:
-        pkl.dump(d_labeled_results, outfile)
+    vision_detector, _ = get_vision_modules()
+    subframes = sample_frames(indices, f'{trailer_path}.mp4')
+    detections = get_faces(subframes, vision_detector)
+    d_annotated_results = annotate_results(detections, subframes, indices)
+    with open(f'{trailer_path}_annotated.pkl', 'wb') as outfile:
+        pkl.dump(d_annotated_results, outfile)
     outfile.close()
+
+
+def evaluate_trailer(trailer_directory:str, evaluation_type:str = 'binary') -> float:
+    trailer_path = os.path.join('evaluation_trailers', trailer_directory, trailer_directory)
+    vision_detector, vision_classifier = get_vision_modules()
+    with open(f'{trailer_path}_annotated.pkl', 'rb') as infile :
+        annotations = pkl.load(infile)
+    infile.close()
+    indices = sorted([int(key.split('_')[1]) for key in annotations])
+    subframes = sample_frames(indices, f'{trailer_path}.mp4')
+    detections = get_faces(subframes, vision_detector)
+    frame_scores = []
+    for i in range(len(subframes)) :
+        frame_annotations = annotations[f'frame_{indices[i]}']
+        frame_detections = [det for det in detections if det['frame_id'] == i]
+        annotated_centroids = [frame_annotations[k]['face_centroid'] for k in range(len(frame_annotations))]
+        mask, matches = [], {}
+        for j in range(len(frame_detections)) :
+            x1, y1, x2, y2 = frame_detections[j]['bbox']
+            det_centroid = ((x2 + x1) / 2, (y2 + y1) / 2)
+            dists = [np.abs(det_centroid[0] - annotated_centroids[k][0]) + np.abs(det_centroid[1] - annotated_centroids[k][1]) 
+                     for k in range(len(annotated_centroids))]
+            mindist, mindist_id = min(dists), np.argmin(dists)
+            if mindist < 3 :
+                matches[j] = mindist_id
+                mask.append(1)
+            else :
+                mask.append(0)
+        frame_detections = [frame_detections[k]['cropped_face'] for k in range(len(frame_detections)) if mask[k] == 1]
+        ages, genders, ethnicities, age_confs, gender_confs, ethnicity_confs = vision_classifier.predict_age_gender_ethnicity(frame_detections, expose_confs=True)
+        
+        for j in range(len(matches)):
```

Comment:
```suggestion
        for key in matches: 
```
Certaines clés sautent à cause de la condition `if mindist < 3 :`. On peut éventuellement renommer `key` en quelque chose de plus explicite mais je suis pas sûr d'avoir bien compris ce que c'est.

Reply:
```suggestion
        for key, j in zip(matches, range(len(ages))):
            frame_annotations[matches[key]]['predicted_age'] = ages[j]
            frame_annotations[matches[key]]['predicted_age_confs'] = age_confs[j]        
            frame_annotations[matches[key]]['predicted_gender'] = genders[j]
            frame_annotations[matches[key]]['predicted_gender_confs'] = gender_confs[j]        
            frame_annotations[matches[key]]['predicted_ethnicity'] = ethnicities[j]
            frame_annotations[matches[key]]['predicted_ethnicity_confs'] = ethnicity_confs[j]
```
Ok en fait le problème qu'on va avoir c'est que matches et les prédictions (par exemple ages) ont la même taille mais matches étant un dictionnaire, il n'aura pas forcément la même indexation que les prédictions. Donc je te propose cette version (pas très belle) comme 'quick fix'. Sinon tu peux revoir la structure de données utilisée pour matches ou tout simplement changer l'indexation des clés de ton dictionnaire.